src/lib/query/ipvlan.rs

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashMap;
+
+use netlink_packet_route::link::{InfoData, InfoIpVlan};
+use serde::{Deserialize, Serialize};
+
+use crate::{Iface, IfaceType};
+
+const IPVLAN_MODE_L2: u16 = 0;
+const IPVLAN_MODE_L3: u16 = 1;
+const IPVLAN_MODE_L3S: u16 = 2;
+
+const IPVLAN_F_PRIVATE: u16 = 0x01;
+const IPVLAN_F_VEPA: u16 = 0x02;
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone, Copy)]
+#[serde(rename_all = "lowercase")]
+#[non_exhaustive]
+pub enum IpVlanMode {
+    L2,
+    L3,
+    L3S,
+    Other(u16),
+    Unknown,
+}
+
+impl Default for IpVlanMode {
+    fn default() -> Self {
+        IpVlanMode::Unknown
```

Comment:
The default should be `IPVLAN_MODE_L3` as kernel code `ipvlan_link_new()` indicate so.

Reply:
Done, thanks!